### ISSUE TITLE
Relations should be defined lazily

### ISSUE THREAD

**Issue by fabien** [strongloop/loopback-datasource-juggler#231]
This should prevent race-conditions. See also: https://github.com/strongloop/loopback-boot/issues/35

While this should already be the case, it turns out not to be fully the case:

```
Uncaught Error: Could not find "notes" relation for Post
      at lookupModelTo (/path-to/project/node_modules/loopback-datasource-juggler/lib/relation-definition.js:475:13)
```

Note loads after Post, in /models/*.js throwing an exception.

**Comment by superkhau** [comment]
@fabien Has the issue been resolved? Can I close it?

**Comment by loay** [comment]
Closing due to inactivity. If the discussion needs to go further, please open the issue. Thanks. 
closes #231
